### PR TITLE
Adds validation for too far future dates and dates in the past, typos…

### PR DESCRIPTION
**Pull request by DevChima**
What this includes:
Date validation for clinic visit dates that are far in the future or dates that have passed
Age display (was returning null because we changed the rapidpro field name)
SMS send flow start
Copy changes

### CODE REVIEW

File: go-app-ussd_registration.js [hunk 1]
```diff
@@ -578,11 +580,24 @@ go.app = (function() {
       return new FreeText(name, {
         question: get_content(name).context(),
         check: function(content) {
+          var givendate = content;
```
Comment (erikh360) [inline go-app-ussd_registration.js]: This variable looks unnecessary, you could just rename content to givendate above.

File: go-app-ussd_registration.js [hunk 2]
```diff
@@ -578,11 +580,24 @@ go.app = (function() {
...
+          givendate = new Date(givendate);
+          var today = new Date();
+          today.setHours(0,0,0,0);
+          var timestamp = new Date().getTime() + (400 * 24 * 60 * 60 * 1000);
```
Comment (erikh360) [inline go-app-ussd_registration.js:587]: I'm not sure what we are doing here? can we add a comment

**Review by erikh360 — COMMENTED** [review]
Looks good just the two comments

File: go-app-ussd_registration.js (hunk 2 above)
Comment (DevChima) [reply go-app-ussd_registration.js:587]: In the rapidpro flow we check if the next clinic visit date entered is more than 400 days from today. If true we return that error. So I was just setting the date I used to compare 400 days from today.

File: go-app-ussd_registration.js (hunk 1 above)
Comment (DevChima) [reply go-app-ussd_registration.js]: Alright

**Review by erikh360 — APPROVED** [review]

**Pull request merged**